be/src/exec/pipeline/scan/morsel.h

```diff
@@ -558,7 +558,15 @@ class DynamicMorselQueue final : public MorselQueue {
     explicit DynamicMorselQueue(Morsels&& morsels) {
         append_morsels(std::move(morsels));
         _size = _num_morsels = _queue.size();
+        _degree_of_parallelism = _num_morsels;
     }
+
+    explicit DynamicMorselQueue(Morsels&& morsels, size_t degree_of_parallelism) {
```

Comment:
just add interface `set_max_degres_of_parallelism`, no need to create a new ctor.